system/Encryption/Handlers/BaseHandler.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/**
+ * CodeIgniter
+ *
+ * An open source application development framework for PHP
+ *
+ * This content is released under the MIT License (MIT)
+ *
+ * Copyright (c) 2014-2017 British Columbia Institute of Technology
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @package    CodeIgniter
+ * @author     CodeIgniter Dev Team
+ * @copyright  2014-2017 British Columbia Institute of Technology (https://bcit.ca/)
+ * @license    https://opensource.org/licenses/MIT	MIT License
+ * @link       https://codeigniter.com
+ * @since      Version 4.0.0
+ * @filesource
+ */
+
+namespace CodeIgniter\Encryption\Handlers;
+
+use Psr\Log\LoggerAwareTrait;
+
+/**
+ * Base class for encryption handling
+ */
+abstract class BaseHandler implements \CodeIgniter\Encryption\EncrypterInterface
+{
+
+	use LoggerAwareTrait;
+
+	/**
+	 * Configuraiton passed from encryption manager
+	 *
+	 * @var string
+	 */
+	protected $config;
+
+	/**
+	 * Logger instance to record error messages and warnings.
+	 *
+	 * @var \PSR\Log\LoggerInterface
+	 */
+	protected $logger;
+
+	//--------------------------------------------------------------------
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $config
+	 */
+	public function __construct($config = [])
+	{
+		$this->logger = \Config\Services::logger(true);
+
+		if (empty($config))
+		{
+			throw new EncryptionException('Encryption handler needs configuration parameters.');
+		}
+		$this->config = $config;
+
+		// make the parameters conveniently accessible
+		foreach ($this->config as $pkey => $value)
+		{
+			$this->$pkey = $value;
+		}
+	}
+
+	// --------------------------------------------------------------------
+
+	/**
+	 * Byte-safe strlen()
+	 *
+	 * @param  string $str
+	 * @return integer
+	 */
+	protected static function strlen($str)
```

Comment:
Why do these methods exist in the different classes? I don't see them being used anywhere.

Reply:
they are used in the extending handlers, eg OpenSSLHandler